packages/guides/src/Nodes/DocumentTree/DocumentEntryNode.php

```diff
@@ -38,27 +36,17 @@ public function getTitle(): TitleNode
         return $this->titleNode;
     }
 
-    public function addChild(DocumentEntryNode $child): void
+    public function addChild(DocumentEntryNode|ExternalEntryNode $child): void
     {
         $this->entries[] = $child;
     }
 
-    /** @return DocumentEntryNode[] */
+    /** @return array<DocumentEntryNode|ExternalEntryNode> */
```

Comment:
Here is an issue, we are stretching the return type. What could help is to add an argument to the getChildren method to specify the filtered child types. Like this:

```
getChildren(string|null $childType = DocumentEntryNode::class) {
```

if we specify null, we can still get all children which we can use internally. Otherwise it will fall back to the original behavior and just provide you an array of `DocumentEntryNode`s. We could directly deprecate this filtering and mention that we will remove this in v2. 


Another approach would be to make sure `ExternalEntryNode has exactly the same interface as the `DocumentEntryNode`   But that would require you to add the methods `getSections`, `getFile` and `isRoot` to `EnternalEntryNode`. Which will be odd, as it does not work like that :-)

Reply:
On investigation it might also make sense to keep getChildren as is this way we dont have to filter in places where we only need the child documents. 

And external menu entries are not really child documents anyway

Reply:
done